libs/react/shelter/src/lib/pages/shelter/HeroCarousel.tsx

```diff
@@ -9,21 +9,24 @@ type TProps = {
 
 export function HeroCarousel(props: TProps) {
   const { shelter, className } = props;
-
   const images = [
     ...(shelter.exteriorPhotos || []),
     ...(shelter.interiorPhotos || []),
   ];
 
-  const imageUrls = images.map((i) => i.file.url);
-
   const parentCss = ['bg-white', 'h-[200px]', className];
   const placeholderCss = ['h-[250px]', className];
 
   if (!images.length) {
     return <ImagePlaceholder className={mergeCss(placeholderCss)} />;
   }
 
+  const rest = images
+    .map((i) => i.file?.url)
+    .filter((u): u is string => !!u && u !== shelter.heroImage);
+
+  const imageUrls = shelter.heroImage ? [shelter.heroImage, ...rest] : rest;
```

Comment:
this keeps typescript happy. The way the backend is designed, if the shelter has any images, `heroImage` can't be null, even if no hero image was explicitly set. But typescript doesn't know that, since `heroImage` optional